cpp/tensorrt_llm/common/stringUtils.cpp

```diff
@@ -26,14 +26,16 @@
 namespace tensorrt_llm::common
 {
 
-void fmtstr_(char const* format, fmtstr_allocator alloc, void* target, va_list args)
+void fmtstr_(char const* format, fmtstr_allocator alloc, void* target, va_list_guard& args_guard)
 {
     va_list args0;
-    va_copy(args0, args);
+    va_copy(args0, args_guard.get());
+    va_list_guard args0_guard(args0);
 
     size_t constexpr init_size = 2048;
     char fixed_buffer[init_size];
-    auto const size = std::vsnprintf(fixed_buffer, init_size, format, args0);
+    auto const size = std::vsnprintf(fixed_buffer, init_size, format, args0_guard.get());
```

Comment:
As a "guard" instead of a "holder", `.get()` from it is not idiomatic. Please revert to pass `va_list args` directly and use the guard object solely to ensure `va_end()` will be called.

Your reply sounds very AI to me. I'm not opposed to AI usage, but if these changes are produced by AI, please review the proposed change yourself first.